shell.nix

```diff
@@ -35,8 +48,62 @@ in pkgs.mkShell {
   LANGUAGE = "en_US.UTF-8";
 
   QTDIR = qtCustom;
+# TODO: still needed?
+# https://github.com/NixOS/nixpkgs/pull/109649
+  QT_INSTALL_PLUGINS = "${qtCustom}/${pkgs.qt515.qtbase.qtPluginPrefix}";
 
   shellHook = ''
-    export MAKEFLAGS="-j$NIX_BUILD_CORES"
-  '';
+    export PATH="${pkgs.lddWrapped}/bin:$PATH"
+    '';
+
+# Used to workaround missing lib links in qt-custom
+# TODO:check again after Qt upgrade
+  LIBRARY_PATH = with pkgs.qt515; pkgs.lib.makeLibraryPath [
+    qtdeclarative
+    qtquickcontrols
+    qtquickcontrols2
+    qtsvg
+    qtmultimedia
+    qtwebview
+    qtwebchannel
+  ];
+
+# Used for linuxdeployqt
+# TODO:check again after Qt upgrade
+  LD_LIBRARY_PATH = with pkgs; lib.makeLibraryPath [
+    qt515.qtquickcontrols2
+    qt515.qtdeclarative
+    qt515.qtmultimedia
+    qt515.qtbase
+    qt515.qtsvg
+    qt515.qtwebengine
+    qt515.qtlocation
+    libglvnd
+    gcc-unwrapped
+    libpulseaudio
+    glib
+    alsaLib
+    expat
+    fontconfig
+    freetype
+    gmp
+    gst_all_1.gst-plugins-base
+    gst_all_1.gstreamer
+    harfbuzz
+    libkrb5
+    libpng
+    libxkbcommon
+    p11-kit
+    xorg.libICE
+    xorg.libSM
+    xorg.libXrender
+    xorg.xcbutilimage
+    xorg.xcbutilkeysyms
+    xorg.xcbutilrenderutil
+    xorg.xcbutilwm
+    zlib
+    xorg.libX11
+    xorg.libxcb
+    xorg.xcbutil
```

Comment:
Same here and in other places, no need for separate lines and just use `with xorg;`.